pulp_rpm/app/serializers/package.py

```diff
@@ -290,6 +268,18 @@ def deferred_validate(self, data):
         data.update(new_pkg)
         return data
 
+    def retrieve(self, data):
+        try:
+            pkg = Package.createrepo_to_dict(read_crpackage_from_artifact(data["artifact"]))
+        except OSError:
+            log.info(traceback.format_exc())
+            raise NotAcceptable(detail="RPM file cannot be parsed for metadata")
+
+        attrs = {key: pkg[key] for key in Package.natural_key_fields()}
```

Comment:
Would a `Package.q()` serve this?

Reply:
I'm not sure I follow, what exact line should it replace?

Reply:
`Package.q()` would produce a Django query object that should be able to replace all the arguments to `Package.objects.filter(...)`

Reply:
I have tried that but I guess I still don't understand this correctly

Reply:
It might help to look at the definition of `q()`.  It's doing pretty much the same thing.

https://github.com/pulp/pulpcore/blob/main/pulpcore/app/models/content.py#L130-L136

Reply:
Are you really sure this should work? I tried many times, and it says Package has no attribite .Q()

Reply:
And if I try through django.db.models.Q(), then Q object has no attribute .touch()

Reply:
It's `q()`.

Reply:
``q()`` is a function which returns query for self.pk, which means it can be called for an object whose PK we already know - how would this be useful then?